src/maketype.jl

```diff
@@ -1,72 +1,127 @@
+        params_to_ints::OrderedDict{Symbol,Int}
+        scale_noise::Symbol
     end)
+
     # Make the default constructor
     constructorex = :($(name)(;
-                  $(Expr(:kw,:f,f)),
-                  $(Expr(:kw,:f_func,f_func)),
-                  $(Expr(:kw,:g,g)),
-                  $(Expr(:kw,:g_func,g_func)),
-                  $(Expr(:kw,:jumps,jumps)),
-                  $(Expr(:kw,:regular_jumps,regular_jumps)),
-                  $(Expr(:kw,:jump_rate_expr,jump_rate_expr)),
-                  $(Expr(:kw,:jump_affect_expr,jump_affect_expr)),
-                  $(Expr(:kw,:p_matrix,p_matrix)),
-                  $(Expr(:kw,:f_symfuncs,f_symfuncs)),
-                  $(Expr(:kw,:syms,syms)),
-                  $(Expr(:kw,:params,params)),
-                  $(Expr(:kw,:symjac,symjac)),
-                  $(Expr(:kw,:reactions,reactions))) =
-                  $(name)(
-                      f,
-                      f_func,
-                      f_symfuncs,
-                      g,
-                      g_func,
-                      jumps,
-                      regular_jumps,
-                      jump_rate_expr,
-                      jump_affect_expr,
-                      p_matrix,
-                      syms,
-                      params,
-                      symjac,
-                      reactions
-                      )) |> esc
-
-                      #f_funcs,symfuncs,pfuncs,syms,symjac) |> esc
+        $(Expr(:kw,:f,f)),
+        $(Expr(:kw,:f_func,f_func)),
+        $(Expr(:kw,:g,g)),
+        $(Expr(:kw,:g_func,g_func)),
+        $(Expr(:kw,:jumps,jumps)),
+        $(Expr(:kw,:regular_jumps,regular_jumps)),
+        $(Expr(:kw,:jump_rate_expr,jump_rate_expr)),
+        $(Expr(:kw,:jump_affect_expr,jump_affect_expr)),
+        $(Expr(:kw,:p_matrix,p_matrix)),
+        $(Expr(:kw,:f_symfuncs,f_symfuncs)),
+        $(Expr(:kw,:syms,syms)),
+        $(Expr(:kw,:params,params)),
+        $(Expr(:kw,:symjac,symjac)),
+        $(Expr(:kw,:reactions,reactions)),
+        $(Expr(:kw,:syms_to_ints, syms_to_ints)),
+        $(Expr(:kw,:params_to_ints, params_to_ints)), 
+        $(Expr(:kw,:scale_noise, Meta.quot(scale_noise)))) =
+        $(name)(
+            f,
+            f_func,
+            f_symfuncs,
+            g,
+            g_func,
+            jumps,
+            regular_jumps,
+            jump_rate_expr,
+            jump_affect_expr,
+            p_matrix,
+            syms,
+            params,
+            symjac,
+            reactions,
+            syms_to_ints,
+            params_to_ints,
+            scale_noise
+            )) |> esc
+
 
     # Make the type instance using the default constructor
     typeex,constructorex
 end
+
+function gen_ode!(rn::DiffEqBase.AbstractReactionNetwork)
+    @unpack reactions, syms_to_ints, params_to_ints = rn
+
+    f_expr        = get_f(reactions, syms_to_ints)
+    rn.f          = eval(make_func(f_expr, syms_to_ints, params_to_ints))
```

Comment:
evaling a function definition like this in a function seems like it's always going to be prone to world age problems. I am wondering if this strategy only could work well in global scope macros.

Reply:
I've been wondering the same. That said, we don't seem to see any problems in tests when we just define functions (ie the ODE and SDE function fields). Where the problem seems to occur is with the types that are instantiated and wrap functions, i.e. `ConstantRateJumps`. 

I tried making a macro, but 
1. Couldn't figure out how to get it to actually construct the jumps without calling `eval` within it too.
2. Even getting the jumps to be defined in the top level by using `Base.eval(@__MODULE__,jumps)` I still got world age errors.

One other note, using function wrappers for `ConstantRateJumps`, i.e. `DirectFW`, gets rid of the test failure.

Reply:
Also, correct me if I'm wrong, but doesn't `ModelingToolkit` do something similar in creating `ODEFunctions` from a `DiffEqSystem`?

Reply:
ModelingToolkit.jl does have an eval form for using it directly, but it's generally made to be used as the IR inside of a macro. I.e., it's made so that macros like the reaction DSL to do its internal calculations, and that wouldn't need eval.

Reply:
Got it. It seems like it may have reached the point that I should try to cook up a MWE and see if anyone has suggestions on Discourse.